### PR TITLE
feat(ime_pinyin): add API to support 9-key input mode

### PR DESCRIPTION
**Pull request by 100ask**
### Description of the feature or fix

A clear and concise description of what the bug or new feature is.

### Checkpoints
- [ ] Follow the [styling guide](https://github.com/lvgl/lvgl/blob/master/docs/CODING_STYLE.md)
- [ ] Run `code-format.py` from the `scripts` folder. [astyle](http://astyle.sourceforge.net/install.html) needs to be installed.
- [ ] Update the documentation

### CODE REVIEW

**Comment by 100ask** [comment]
I hope 9-key input mode can be released with v8.3, so I finish 9-key input mode today.
The following is the operation effect:

2022年7月5日-lv_ime_pinyin_v2

File: src/extra/others/ime/lv_ime_pinyin.c [hunk 1]
```diff
@@ -429,6 +450,30 @@ void lv_ime_pinyin_set_dict(lv_obj_t * obj, lv_pinyin_dict_t * dict)
     init_pinyin_dict(obj, dict);
 }
 
+/**
+ * Set mode, 26 key input(k26) or 9 key input(k9).
+ * @param obj  pointer to a Pinyin input method object
+ * @param en   1: 26 key input(k26), 0: key input(k9)
+ */
+void lv_ime_pinyin_set_mode(lv_obj_t * obj, bool en)
```
Comment (kisvegabor) [inline src/extra/others/ime/lv_ime_pinyin.c]: I suggest using an `enum` here.

**Review by kisvegabor — COMMENTED** [review]
It looks good! :+1: 

I have only 1 suggestion.

File: src/extra/others/ime/lv_ime_pinyin.c (hunk 1 above)
Comment (YobieZhou) [reply src/extra/others/ime/lv_ime_pinyin.c]: This proposal is very good. I will revise it soon.

File: src/extra/others/ime/lv_ime_pinyin.h [hunk 2]
```diff
@@ -71,6 +91,13 @@ void lv_ime_pinyin_set_keyboard(lv_obj_t * obj, lv_obj_t * kb);
  */
 void lv_ime_pinyin_set_dict(lv_obj_t * obj, lv_pinyin_dict_t * dict);
 
+/**
+ * Set mode, 26-key input(k26) or 9-key input(k9).
+ * @param obj  pointer to a Pinyin input method object
+ * @param mode   the mode from 'lv_keyboard_mode_t'
```
Comment (kisvegabor) [inline src/extra/others/ime/lv_ime_pinyin.h]:
```suggestion
 * @param mode   the mode from 'lv_ime_pinyin_mode_t'
```

**Comment by kisvegabor** [comment]
LGTM, I just fixed a [minor typo](1b61145).

Thank you for the great quality patch again

**Pull request merged**